modify Start-PSPester to accept -Quiet to eliminate Pester output

JamesWTruher opened the pull request:
also modify travis.ps1 to include -Quiet for Pester args to reduce
stdout log size which should hopefully allow daily builds to run
without being cancelled due to too much output

JamesWTruher committed:
modify Start-PSPester to accept -Quiet to eliminate Pester output

also modify travis.ps1 to include -Quiet for Pester args to reduce
stdout log size which should hopefully allow daily builds to run
without being cancelled due to too much output

adityapatwardhan commented:
:shipit:

lzybkr commented:
With `-Quiet`, it looks like Pester will not report any failures to the screen.

So how will failing tests be reported in a way we can easily see?

I see our scripts will throw if there are any errors, but it just reports a count.  If I'm missing the code that would report the errors to the console, can you point me to that code?

JamesWTruher commented:
The log file always contains the errors even if they're not emitted to the screen during the test run. I can add functionality to Test-PSPesterResults to emit the errors before throwing. This is probably the best we can hope for because there is so much output there is no way that Travis-CI will be able to run all of our tests.

lzybkr reviewed (changes requested):
Please modify `Test-PSPesterResults` so that it reports as much detail as possible about the test failures - this will make it easier for the maintainers to quickly see why builds are failing.

JamesWTruher commented:
Do you wish to see the error details at all times? 
Should the default behavior for all test runs be -Quiet?

lzybkr commented:
I can hardly imagine a scenario where I wouldn't want errors logged to the console, so yes, always.

As for `-Quiet` being the default for all runs - maybe just in CI. I have on occasion wanted to check that my test ran in CI, but less spew to the console in CI makes finding the errors easier. If the log files are uploaded, maybe it's a reasonable default for CI.

JamesWTruher commented:
for the moment, I'm going to leave the verbose output for CI runs as I believe that the log output is useful. We can use the daily builds in travis-ci to determine whether we like it or no.

JamesWTruher committed:
Add error detail information as part of Test-PSPesterResults

Test-PSPesterResults will now also emit information about the errors
found during a test run rather than just the number of failed tests.
Created a new errorlog function so the output of errors will be red.

iSazonov commented:
Related PR #2600 from @SteveL-MSFT who open a feature request in Pester repo.

JamesWTruher commented:
@lzybkr requested change should be handled by https://github.com/PowerShell/PowerShell/pull/2795/commits/2441c8937fd7618aba60a6a2f0fcabc76322168d
